src/commands/pruneSystem.ts

```diff
@@ -15,21 +15,26 @@ export async function pruneSystem(context: IActionContext): Promise<void> {
     // no need to check result - cancel will throw a UserCancelledError
     await ext.ui.showWarningMessage(confirmPrune, { modal: true }, { title: 'Remove' });
 
-    /* eslint-disable @typescript-eslint/promise-function-async */
-    const containersResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneContainers(), context);
-    const imagesResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneImages(), context);
-    const networksResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneNetworks(), context);
-    const volumesResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneVolumes(), context);
-    /* eslint-enable @typescript-eslint/promise-function-async */
+    await vscode.window.withProgress(
+        { location: vscode.ProgressLocation.Notification, title: localize('vscode-docker.commands.pruneSystem.pruning', 'Pruning system...') },
+        async () => {
+            /* eslint-disable @typescript-eslint/promise-function-async */
+            const containersResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneContainers(), context);
+            const imagesResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneImages(), context);
+            const networksResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneNetworks(), context);
+            const volumesResult = await callDockerodeWithErrorHandling(() => ext.dockerode.pruneVolumes(), context);
```

Comment:
Not sure if you want to address it with this change, but we got user feedback we should ask for extra confirmation about pruning volumes, as they might contain useful data.

Reply:
I think probably the right way is to add an extra warning to the prompts--that pruning volumes can result in lost data.